File: focus_todo_computer/focus-todo/src/sync/supabaseConfig.ts
export interface SupabaseConfig {
  url: string | null
  anonKey: string | null
}

function normalizeConfigValue(value: string | null | undefined): string | null {
  if (!value) return null
  const trimmed = value.trim()
  return trimmed ? trimmed : null
}

export const SUPABASE_URL = normalizeConfigValue(
  import.meta.env.VITE_SUPABASE_URL,
)

export const SUPABASE_ANON_KEY = normalizeConfigValue(
  import.meta.env.VITE_SUPABASE_ANON_KEY,
)

export function getSupabaseConfig(): SupabaseConfig {
  return {
    url: SUPABASE_URL,
    anonKey: SUPABASE_ANON_KEY,
  }
}

export function hasSupabaseConfig(config: SupabaseConfig = getSupabaseConfig()): boolean {
  return Boolean(config.url && config.anonKey)
}

File: focus_todo_mobile/lib/sync/supabase_config.dart
import 'package:supabase_flutter/supabase_flutter.dart';

class SupabaseConfig {
  const SupabaseConfig({required this.url, required this.anonKey});

  final String? url;
  final String? anonKey;
}

String? _normalizeConfigValue(String value) {
  final trimmed = value.trim();
  return trimmed.isEmpty ? null : trimmed;
}

const String _rawSupabaseUrl = String.fromEnvironment(
  'SUPABASE_URL',
  defaultValue: '',
);
const String _rawSupabaseAnonKey = String.fromEnvironment(
  'SUPABASE_ANON_KEY',
  defaultValue: '',
);

final String? SUPABASE_URL = _normalizeConfigValue(_rawSupabaseUrl);
final String? SUPABASE_ANON_KEY = _normalizeConfigValue(_rawSupabaseAnonKey);
bool _supabaseInitialized = false;

SupabaseConfig getSupabaseConfig() {
  return SupabaseConfig(url: SUPABASE_URL, anonKey: SUPABASE_ANON_KEY);
}

bool hasSupabaseConfig([SupabaseConfig? config]) {
  final resolved = config ?? getSupabaseConfig();
  return resolved.url != null && resolved.anonKey != null;
}

bool isSupabaseInitialized() => _supabaseInitialized;

Future<void> initializeSupabaseIfConfigured([SupabaseConfig? config]) async {
  final resolved = config ?? getSupabaseConfig();
  if (_supabaseInitialized || !hasSupabaseConfig(resolved)) {
    return;
  }

  await Supabase.initialize(url: resolved.url!, anonKey: resolved.anonKey!);
  _supabaseInitialized = true;
}
